halpybot/commands/caseutils.py

```diff
@@ -58,41 +97,555 @@ async def cmd_go(ctx: Context, args: List[str]):
+    else:
+        list_filter = None
+
+    filtered_cases = [
+        case
+        for case in caseboard.values()
+        if (not list_filter)
+        or (
+            list_filter
+            in (
+                case.platform.name.casefold().replace("_horizons", ""),
+                case.case_type.name.casefold(),
+            )
+        )
+    ]
+
+    message = "Here's the current case board:\n\n"
+    for case in filtered_cases:
+        hskf = (
+            "Seal"
+            if case.case_type in (CaseType.SEAL, CaseType.BLUE, CaseType.BLACK)
+            else "Fisher"
+            if case.case_type == CaseType.FISH
+            else "Unknown"
+        )
         long_ago = now(tz="utc").diff(case.updated_time).in_words()
         plt = case.platform.name.replace("_", " ")
         message += (
             f"Case {case.board_id}: Client: {case.client_name}, Platform: {plt}, "
             f"Type: {hskf}, Status: {case.status.name}, Updated: {long_ago} ago.\n"
         )
-    return await ctx.redirect(f"{message}{len(caseboard)} Cases on the Board.")
+
+    message += f"\n{len(caseboard)} Cases on the Board."
+    if list_filter:
+        message += (
+            f" Showing {len(filtered_cases)} that match(es) the filter {list_filter!r}."
+        )
+    return await ctx.redirect(message)
 
 
 @Commands.command("listcase")
-async def cmd_listcase(ctx: Context, args: List[str]):
+@Require.permission(Drilled)
+@CaseUtils.gather_case(1)
+async def cmd_listcase(ctx: Context, args: List[str], case: Case):
     """
     Send a user the key details of a case on the board in DMs
 
     Usage: !listcase [board ID]
     Aliases: n/a
     """
-    if not args:
-        return await ctx.reply(get_help_text(ctx.bot.commandsfile, "listcase"))
+    return await ctx.redirect(f"{case}")
+
+
+# CASE MANAGEMENT
+@Commands.command("rename")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_renamecase(ctx: Context, args: List[str], case: Case):
+    """
+    Rename the user of an active case
+
+    Usage: !rename [board ID] [new name]
+    Aliases: n/a
+    """
+    try:
+        await ctx.bot.board.rename_case(args[1], case, ctx.sender)
+    except (AssertionError, ValueError) as err:
+        return await ctx.reply(str(err))
+    for channel in config.channels.rescue_channels:
+        await ctx.bot.message(
+            channel,
+            f"Client for case {case.board_id} set to {args[1]!r} from {case.client_name!r}",
+        )
+
+
+@Commands.command("ircn")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_ircn(ctx: Context, args: List[str], case: Case):
+    """
+    Rename the user of an active case
+
+    Usage: !ircn [board ID] [valid IRC user]
+    Aliases: n/a
+    """
+    try:
+        await User.get_info(ctx.bot, args[1])
+    except (AttributeError, NoUserFound):
+        return await ctx.reply("That's not an IRC user!")
```

Comment:
I am not convinced you should be catching `AttributeError` here.

Reply:
Required as AttrE is thrown if a given target doesn't exist for Pydle to look up.